master.js

```diff
@@ -0,0 +1,151 @@
+let data = [];
+let query = (document.querySelector("input:first-child:last-child").value).toLowerCase()
+let isLoading = true;
+
+async function fetchData (){
+    const response = await fetch("https://dummyjson.com/todos")
+    const {todos} = await response.json()
+    data = todos;
+    if(window.localStorage.getItem("list"))
+        data = JSON.parse(window.localStorage.getItem("list"));
+}
+
+
+fetchData().then(() => {
+    isLoading = false;
+    fillTable(data)
+})
+
+function fillTable(givenData){
+    let loader = document.querySelector(".loading");
+    if(isLoading)
+    null
+else 
+   {
+    let filterdData = givenData.filter((obj)=>{
+    return obj.todo.toLowerCase().includes(query)
+    }) 
+    loader.style.display = "none";
+     let tbody = document.querySelector("table tbody")
+     filterdData.forEach((obj)=>{
+        // console.log(obj)
+        let id = obj.id;
+        let description = obj.todo;
+        let userID = obj.userId;
+        let status = obj.completed ? "completed" : "pending"
+        let tableRow = document.createElement("tr")
+        let arr = [id, description, userID, status];
+        for(let i=0; i<4; i++){
+            let td = document.createElement("td");
+            let text = document.createTextNode(arr[i])
+            td.appendChild(text);
+            if(arr[i] == description){
+                td.classList.add("description")
+                td.setAttribute("title", description)
+            }
+            tableRow.appendChild(td)
+        }
+        let deleteBtn = document.createElement("button");
+        deleteBtn.id = "delete"
+        deleteBtn.dataset.id = obj.id;
+        deleteBtn.appendChild(document.createTextNode("Delete"))
+        let doneBtn = document.createElement("button");
+        doneBtn.id = "done"
+        doneBtn.dataset.id = obj.id;
+        if(obj.completed)
+            doneBtn.appendChild(document.createTextNode("Undo"))
+        else doneBtn.appendChild(document.createTextNode("Done")) 
+        let td = document.createElement("td");
+        td.append(deleteBtn, doneBtn)
+        tableRow.appendChild(td)
+        obj.completed ? tableRow.style.background = "#cccccc3d": null
+        tbody.appendChild(tableRow)
```

Comment:
you are updating the DOM everytime you face new item, can't you append them all in one shot?